Server/middleware.ts

```diff
@@ -48,16 +48,17 @@ async function isStaff (ctx: ParameterizedContext, next: Next): Promise<void> {
 function hasRole (section: string, role: string) {
     return async (ctx: ParameterizedContext, next: Next): Promise<void> => {
         const member = await getMember(ctx.state.user.discord.userID);
-        if (
-            member && 
-            (
-                member.roles.cache.has(config.discord.roles[section][role]) || 
+        if (member) {
+            const hasRole =  Array.isArray(config.discord.roles[section][role]) ? config.discord.roles[section][role].some(r => member.roles.cache.has(r)) : member.roles.cache.has(config.discord.roles[section][role]);
+            if (
+                hasRole || 
+                member.roles.cache.has(config.discord.roles.corsace.core) || 
                 member.roles.cache.has(config.discord.roles.corsace.corsace) || 
-                (role === "corsace" ? false : member.roles.cache.has(config.discord.roles.corsace.headStaff))
-            )
-        ) {
-            await next();
-            return;
+                (role === "corsace" || role === "core" ? false : config.discord.roles.corsace.headStaff.some(r => member.roles.cache.has(r)))
```

Comment:
this would be more clear with (another) nested if imo

Reply:
i split it to parallel hasRoles